source/js/RecipeInfo.js

```diff
@@ -0,0 +1,283 @@
+      } else {
+        info.removeChild(showIngredients);
+        ingredients.textContent = "Show Ingredients";
+      }
+    });
+    const nutrition = document.createElement("button");
+    nutrition.classList.add("button");
+    nutrition.textContent = "Show Nutritions";
+
+    const saveRecipe = document.createElement("button");
+    saveRecipe.classList.add("button");
+    saveRecipe.textContent = this.isSaved ? "Unsave Recipe" : "Save Recipe";
+    saveRecipe.addEventListener("click", () => {
+      if (!this.isSaved) {
+        this.addRecipeToSaved(this.url).then(() => {
+          saveRecipe.textContent = "Unsave Recipe";
+        });
+      } else {
+        this.removeRecipeFromSaved(this.url).then(() => {
+          saveRecipe.textContent = "Save Recipe";
+        });
+      }
+    });
+
+    info.appendChild(ingredients);
+    info.appendChild(nutrition);
+    info.appendChild(saveRecipe);
+
+    this.shadowRoot.appendChild(styleElem);
+    this.shadowRoot.appendChild(info);
+  }
+}
+
+customElements.define("recipe-info", RecipeInfo);
+
+/*********************************************************************/
+/***                       Helper Functions:                       ***/
+/***          Shout out to the TA's lemme just yoink these         ***/
+/*********************************************************************/
+
+/**
+ * Recursively search for a key nested somewhere inside an object
+ * @param {Object} object the object with which you'd like to search
+ * @param {String} key the key that you are looking for in the object
+ * @returns {*} the value of the found key
+ */
+function searchForKey(object, key) {
+  var value;
+  Object.keys(object).some(function (k) {
+    if (k === key) {
+      value = object[k];
+      return true;
+    }
+    if (object[k] && typeof object[k] === "object") {
+      value = searchForKey(object[k], key);
+      return value !== undefined;
+    }
+  });
+  return value;
+}
+
+/**
+ * Similar to getUrl(), this function extracts the organizations name from the
+ * schema JSON object. It's not in a standard location so this function helps.
+ * @param {Object} data Raw recipe JSON to find the org string of
+ * @returns {String} If found, it retuns the name of the org as a string, otherwise null
+ */
+function getOrganization(data) {
+  if (data.publisher?.name) return data.publisher?.name;
+  if (data["@graph"]) {
+    for (let i = 0; i < data["@graph"].length; i++) {
+      if (data["@graph"][i]["@type"] == "Organization") {
+        return data["@graph"][i].name;
+      }
+    }
+  }
+  return null;
+}
+
+/**
+ * Similar to getOrganization(), this extracts recipe name from raw JSON
+ * @param {Object} Data Raw recipe JSON to find name of
+ * @returns {String} if found, returns the name of recipe as string, otherwise null
+ */
+function getRecipeTitle(data) {
+  if (data.name) return data.name;
+  let value = null;
+  if (data["@graph"]) {
+    data["@graph"].forEach((obj) => {
```

Comment:
Codacy has a fix for the issue: [There must be a hyphen before @param description.](https://app.codacy.com/gh/cse110-fa21-group27/cse110-fa21-group27/pullRequest?prid=8447291)
```suggestion
 * @param {Object} data - Raw recipe JSON to find the URL of
```